test/util_api.go

```diff
@@ -15,3 +19,29 @@ func CheckRespStatus(c *check.C, resp *http.Response, status int) {
 		c.Assert(resp.StatusCode, check.Equals, status, check.Commentf("Response Body: %v", string(body)))
 	}
 }
+
+// GetMetricValue get the specific metrics from /metrics endpoint,
+// if cannot find this metrics, return not found.
+func GetMetricValue(c *check.C, key string) (float64, bool) {
+	var (
+		val   float64
+		found bool
+	)
+
+	resp, err := request.Get("/metrics")
+	c.Assert(err, check.IsNil)
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	c.Assert(err, check.IsNil)
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		if strings.Contains(line, key) {
```

Comment:
How about:
```
if !strings.Contains(line, key){
    continue
}
data := strings.Split(line, " ")
if len(data) < 2 {
    c.Fatal("blabla")
}
val, err = strconv.ParseFloat(data[1], 64)
c.Assert(err, check.IsNil)
return val, true
```

Reply:
We don't need to check the length of the line. It will always  be 2.

Reply:
What if someone mocks a `/metrics` endpoint and returns the data with incorrect format. The key point is that we can not guarantee what a unknown backend does and we must take every possibility into consideration and make our own code more robust. WDYT